app/src/main/java/org/simple/clinic/encounter/Encounter.kt

```diff
@@ -33,6 +36,24 @@ data class Encounter(
   interface RoomDao {
 
     @Insert(onConflict = REPLACE)
-    fun save(encounters: List<Encounter>)
+    fun save(encounters: List<Encounter>): Completable
+
+    @Query("UPDATE Encounter SET syncStatus = :newStatus WHERE syncStatus = :oldStatus")
+    fun updateSyncStatus(oldStatus: SyncStatus, newStatus: SyncStatus): Completable
+
+    @Query("UPDATE Encounter SET syncStatus = :newStatus WHERE uuid IN (:uuids)")
+    fun updateSyncStatus(uuids: List<UUID>, newStatus: SyncStatus): Completable
+
+    @Query("SELECT * FROM Encounter WHERE syncStatus = :syncStatus")
+    fun recordsWithSyncStatus(syncStatus: SyncStatus): Observable<List<Encounter>>
+
+    @Query("SELECT COUNT(uuid) FROM Encounter")
+    fun recordCount(): Observable<Int>
```

Comment:
Curious, what are we doing with this information?

Reply:
If this information is used to display static data, should we consider using a `Single` instead?

Reply:
Actually this method is not needed by most `SynceableRepository` including this one.
But in order to satisfy the contract, we need to implement this.
We can actually consider removing this altogether method from `Repository` altogether.
Also, we could change the contract to accept `Single` instead of `Observable`.
I will add this as a separate refactoring.

Reply:
Works!